valohai_yaml/objs/deployment.py

```diff
@@ -0,0 +1,41 @@
+from typing import Optional
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.types import SerializedDict
+
+
+class DeploymentDefaults(Item):
+    """Represents the default values to use *if* matching deployment does not exist in the project context."""
+
+    def __init__(
+        self,
+        *,
+        target: str,
+        allow_cors: Optional[bool] = None,
+        allow_configuration_snippets: Optional[bool] = None,
+    ) -> None:
+        self.target = target
+        self.allow_cors = allow_cors
+        self.allow_configuration_snippets = allow_configuration_snippets
```

Comment:
I would maybe not strictly specify the defaults type at all in valohai-yaml, to  give us a little more wiggle room to change things.

`defaults` could just be a `dict[str, Any]` bag, and we'd deal with it in the app.

`target` could be made soft-required by way of a lint warning in `Deployment`.

Reply:
I had it as a dict first but noticed that it was nicer to use with type hints during tests when it was a real class

I... can change it back if you want

how do you see it helping with wiggle room? at least with types you'd know you have wrong version of Valohai YAML in the context you are working in :laughing:

Reply:
I changed it to a `dict` in either case!